### PR TITLE
HT-2586

### PR DESCRIPTION
**Pull request by respinos**
- upgraded `Image::Info`
- consolidated image metadata routines into `Image::Utils`
- added `Image::Info::_patches` to reduce JPEG2000 vs. everything else conditionals
- for images: added support for specifying size as `ppi`, e.g. `size=ppi:150`
- for PDF: determined that transforming bitonal TIFFs is pointless and now excuse them from any downsampling

### CODE REVIEW

File: lib/Image/Info/JPEG2000.pm [hunk 1]
```diff
@@ -6,21 +6,30 @@ sub ImageInfo {
...
     };
-    if ( $$data{'colourSpecificationBox/enumCS'} == 16 ) {
+
+    if ( $$data{'jp2HeaderBox/colourSpecificationBox/enumCS'} == 16 ) {
         $$result{ColorSpace} = qq{sRGB};
-    } elsif ( $$data{'colourSpecificationBox/enumCS'} == 17 ) {
+    } elsif ( $$data{'jp2HeaderBox/colourSpecificationBox/enumCS'} == 17 ) {
         $$result{ColorSpace} = qq{sLUM};
-    } elsif ( $$data{'colourSpecificationBox/enumCS'} == 18 ) {
+    } elsif ( $$data{'jp2HeaderBox/colourSpecificationBox/enumCS'} == 18 ) {
```
Comment (aelkiss) [inline lib/Image/Info/JPEG2000.pm:32]: There should not be any YCC JPEG2000 images in the repository

File: lib/Image/Utils.pm [hunk 2]
```diff
@@ -0,0 +1,92 @@
+package Image::Utils;
```
Comment (aelkiss) [inline lib/Image/Utils.pm:1]: It strikes me that there's significant duplication between this and https://github.com/hathitrust/feed/blob/master/lib/HTFeed/Stage/ImageRemediate.pm -- maybe worth thinking eventually how we could refactor these into a common shared dependency at some point.

File: lib/Image/Info/JPEG2000.pm (hunk 1 above)
Comment (aelkiss) [inline lib/Image/Info/JPEG2000.pm:32]: It wasn't clear to me if this is your code or code copied from `Image::Info` (at least without going digging through `Image::Info`, but FWIW there shouldn't be any JPEG2000 images in the repository with color space `sYCC`)

Comment (respinos) [reply lib/Image/Info/JPEG2000.pm:32]: This is mostly a port of the BoxValidator from jpylyzer.

**Comment by aelkiss** [comment]
Nothing raises any red flags for me.

**Pull request merged**